test/pipelines/pipelines.test.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable func-names */
+import { expect } from 'chai';
+import { closeApplication, createApplication } from '../application';
+import { DEFAULT_GLOBAL_TIMEOUT } from '../constants';
+import {
+  PRIMARY_PIPELINE,
+  SECONDARY_PIPELINE,
+  FIRST_ALGORITHM,
+  SECOND_ALGORITHM,
+  THIRD_ALGORITHM,
+  SECOND_ALGORITHM_INDEX,
+} from '../fixtures/pipelines/pipelines';
+import { mochaAsync } from '../utils';
+import {
+  addPipeline,
+  clickAddButton,
+  addAlgorithmToPipeline,
+  clickSavePipelineButton,
+  checkPipelineRowLayout,
+  clickEditPipelineButton,
+  removeAlgorithmPipeline,
+  clickDeletePipelineButton,
+  getNumberOfPipelines,
+  editNamePipeline,
+  editDescriptionPipeline,
+} from './utils';
+
+describe('Pipelines Scenarios', function () {
+  this.timeout(DEFAULT_GLOBAL_TIMEOUT);
+  let app = null;
+
+  beforeEach(
+    mochaAsync(async () => {
+      app = await createApplication({
+        database: {
+          pipelines: [PRIMARY_PIPELINE, SECONDARY_PIPELINE],
+          algorithms: [FIRST_ALGORITHM, SECOND_ALGORITHM, THIRD_ALGORITHM],
+        },
+        responses: { showMessageDialogResponse: 1 },
+      });
+      const { client } = app;
+      await client.goToPipelines();
+    }),
+  );
+
+  afterEach(() => {
+    return closeApplication(app);
+  });
+
+  it(
+    'Adds, Edits, Deletes a pipeline',
+    mochaAsync(async () => {
+      const { client } = app;
+
+      const nbrPipelinesPrev = await getNumberOfPipelines(client);
+
+      // add pipeline
+      await clickAddButton(client);
+      await addPipeline(client, PRIMARY_PIPELINE);
+
+      await addAlgorithmToPipeline(client, FIRST_ALGORITHM);
+      await addAlgorithmToPipeline(client, SECOND_ALGORITHM);
+      await addAlgorithmToPipeline(client, THIRD_ALGORITHM);
+
+      await clickSavePipelineButton(client);
+      await checkPipelineRowLayout(client, PRIMARY_PIPELINE);
+
+      const nbrPipelinesAfterAdd = await getNumberOfPipelines(client);
+      expect(nbrPipelinesAfterAdd - nbrPipelinesPrev).to.be.equal(1);
```

Comment:
I see you reused how algorithms where tested. This line was defined as such because we previously had a non-defined number of default algorithms at the start of the app (now it is no more the case). In your case you don't particularly need to test the difference before and after an action, since you define the default data. But you can leave it as it is, it's not a problem.